pkg/cli/nodelocal.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2019 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cli
+
+import (
+	"database/sql/driver"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+const (
+	chunkSize = 4 * 1024
+)
+
+var nodeLocalUploadCmd = &cobra.Command{
+	Use:   "upload <source> <destination>",
+	Short: "Upload file from source to destination",
+	Long: `
+Uploads a file to a gateway node's local file system using a SQL connection.
+`,
+	Args: cobra.MinimumNArgs(2),
+	RunE: maybeShoutError(runUpload),
+}
+
+//var verUploadFile = version.MustParse("")
+
+func runUpload(cmd *cobra.Command, args []string) error {
+	conn, err := makeSQLClient("cockroach nodelocal", useSystemDb)
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+	// TODO: which version should we be checking for?
+	//if err := conn.requireServerVersion(verUploadFile); err != nil {
+	//	return err
+	//}
+
+	source := args[0]
+	destination := args[1]
+	reader, err := openSourceFile(source)
+	if err != nil {
+		return err
+	}
+	return uploadFile(conn, reader, destination)
+}
+
+func openSourceFile(source string) (io.ReadCloser, error) {
+	stat, err := os.Stat(source)
```

Comment:
Remove this call and just use `f.Stat()`. Otherwise there's a race condition since the file could change between the call to stat and the call to open.